lib/question.js

```diff
@@ -1117,6 +1117,8 @@ module.exports = {
             urls.baseUrl = urlPrefix;
         }
 
+        urls.workspaceUrl = urlPrefix + '/workspace/' + variant.workspace_id;
```

Comment:
I think we should omit `urls.workspaceUrl` if the variant doesn't have a workspace defined (that is, if `variant.workspace_id` is null). That'll keep other code simpler:

```js
// Complex
if (locals.workspaceUrl.split('/').pop() !== 'null') { }

// Simple!
if (locals.workspaceUrl != null) { }
```

Reply:
Changed in https://github.com/PrairieLearn/PrairieLearn/pull/2665/commits/790c99a52d62c601b1ef491a85e60c16d7fccef6